### DISCUSSION TITLE
Really odd behaviour with the Greaseweazle v4

### DISCUSSION THREAD

**Discussion by MrWoof54** [keirf/greaseweazle#372] [Q&A]
Hi all,

I recently picked up a greaseweazle v4 from Amiga kit, grabbed the software and the greaseweazle GUI for use on a windows 10 laptop. 

I was using a really old floppy drive that came out of a P3 base unit at first, and it was being powered by the PSU in the base unit as well. I had a few issues at the start but started using a floppy ide cable with a twist and this seemed to resolve any issues. I could format a disk, I read some disks into ADF files and I was able to write a copy of workbench back over to a disk. 

I'm trying to go through a pile of old amiga disks to see which are still workable and which are ready for the bin, the process of formatting a disk and then trying to write would work for a while and then I can't read or write, apart from format the disk. That always works, which tells me I've got connectivity end to end. 

I thought it might of been the old floppy drive, so I bought a brand new samsung floppy drive from ebay and I get the same behaviour. It'll work fine for a while and then I just get an error when trying to write, usually a failed to verify track 0.0 and there's a flux error when trying to read a disk into an adf file.

I've upgraded the unit to the 1.4 firmware, I've tried a few IDE cables (no twist, twist), two usb cables, two floppies, two laptops, one floppy powered by the old desktop, new floppy drive, powered off the greaseweazel and I've got the latest software. 

Is this me doing some incorrect and / or am I missing something obvious?

**Comment by keirf** [comment]
So the setup continues to work, but then stops for a while? You'll need to be specific with commands you run, and errors you get, and what appears to "fix" the problems temporarily for you.

**Reply by MrWoof54** [reply]
Hi, I think I've answered my own question and this might help some other people with a similar issue. As the original floppy drive was really old, I would clean the heads of the drive now and then to keep it going. I didn't think with the brand new drive it would be an issue, but I've just cleaned it now and the heads were dirty, after not that many disks really and I suppose it shows how dirty the old floppy disks can get.

I tried to do a write before the clean and it failed with the failed to verify track 0.0, using a very basic command:

gw.exe --time read --diskdefs "C:\greaseweazle-win64\greaseweazle-1.16.1-win64\diskdefs.cfg" --format=amiga.amigados "C:\greaseweazle-win64\greaseweazle-1.16.1-win64\reads\WorkBench.adf"

Writing c=0-79:h=0-1
Format amiga.amigados
T0.0: Writing Track (Flux: 200.0ms period, 219.9 ms total, Terminate at index)
T0.0: Writing Track (Verify Failure: Retry #1)
T0.0: Writing Track (Verify Failure: Retry #2)
T0.0: Writing Track (Verify Failure: Retry #3)
** FATAL ERROR:
Failed to verify Track 0.0
Time elapsed: 5.19 seconds

After the clean, no issues with reading or writing again. It looks like I'll be cleaning the surface of many disks with alcohol and then trying again.

**Reply by keirf** [reply]
Yes this is a common problem with mouldy old disks unfortunately. And if you clean the disks too aggressively you can make them unreadable.

**Reply by MrWoof54** [reply]
yep totally agree, im being careful with earbuds and rotating them round
